server/runtime/src/main/java/org/infinispan/server/configuration/DataSourcesConfigurationBuilder.java

```diff
@@ -0,0 +1,47 @@
+package org.infinispan.server.configuration;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.infinispan.commons.configuration.Builder;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+
+public class DataSourcesConfigurationBuilder implements Builder<DataSourcesConfiguration> {
+
+   private final AttributeSet attributes;
+   private final ServerConfigurationBuilder server;
+
+   private Map<String, DataSourceConfigurationBuilder> dataSources = new LinkedHashMap<>(2);
+
+   DataSourcesConfigurationBuilder(ServerConfigurationBuilder server) {
+      this.server = server;
+      attributes = DataSourcesConfiguration.attributeDefinitionSet();
+   }
+
+   DataSourceConfigurationBuilder dataSource(String name, String jndiName) {
+      DataSourceConfigurationBuilder builder = new DataSourceConfigurationBuilder(server, name, jndiName);
+      dataSources.put(name, builder);
+      return builder;
+   }
+
+   @Override
+   public void validate() {
+   }
```

Comment:
Should we make the `validate` method a default no-op in the interface? Doesn't have to be in this PR.